authentication/azure/auth_test.go

```diff
@@ -146,6 +146,8 @@ func TestGetMSI(t *testing.T) {
 }
 
 func TestFallbackToMSI(t *testing.T) {
+	os.Setenv("MSI_ENDPOINT", "test")
```

Comment:
Please, do not depend on Env variables in unit tests because it is global state that can make the tests flaky if ran in parallel.

Reply:
This is a partial revert of the revert of https://github.com/dapr/components-contrib/pull/1132

This was the easiest way @pkedy was able to get the MSI tests to pass after upgrading the auth libraries. This change was in master until we reverted it (for unrelated reasons) last week.

Reply:
Also keep in mind `MSI_ENDPOINT` or `IDENTITY_ENDPOINT` are special purposes environment variables that are set in Azure Managed Identity environments. See for example https://docs.microsoft.com/en-us/azure/app-service/overview-managed-identity

MSI itself in fact always relies on these special environment variables. As such this use here is fine!

Reply:
The alternative is to remove unit tests for MSI entirely which isn't great.

Reply:
We need to find a solution where tests that depend on the presence or absence of some environment variables are behind a mutex to make sure they don't run in parallel. I will approve this PR but this should be fixed afterwards since it adds flakiness to unit tests.